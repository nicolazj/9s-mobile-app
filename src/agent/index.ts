import axios from 'axios';
import auth, { Auth } from '../states/Auth';
import { ClientConfig } from '../types';
import basicAgent from './basic';
import companyAgent from './company';
import config from './config';
import publicAgent from './public';
import userAgent from './user';

export const APIClient = (config: ClientConfig) => (auth: Auth) => {
  const instance = axios.create({
    baseURL: config.baseURL,
  });

  return {
    get basic() {
      return basicAgent(instance, config, auth);
    },
    get public() {
      return publicAgent(instance, config, auth);
    },
    get user() {
      return userAgent(instance, config, auth);
    },
    get company() {
      return companyAgent(instance, config, auth);
    },
  };
};

export default APIClient(config)(auth);
