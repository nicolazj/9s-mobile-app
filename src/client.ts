import axios from 'axios';
import jwt from 'jwt-decode';
import qs from 'qs';
import { Buffer } from 'buffer';

interface Company {
  companyName: string;
  companyUuid: string;
  industryUuid: String;
}
interface Connection {
  appKey: string;
  createdAt: string;
  createdBy: string;
  id: string;
  status: string;
}
type Client = typeof client;
type UserClient = ReturnType<Client>;

const client = ({
  baseURL,
  tenantId,
  appKey,
  appSecret,
}: {
  baseURL: string;
  tenantId: string;
  appKey: string;
  appSecret: string;
}) => {
  const instance = axios.create({
    baseURL,
  });
  const basicAuthToken = Buffer.from(`${appKey}:${appSecret}`).toString('base64');
  return {
    login: async (username: string, password: string) => {
      const r = await instance.post(
        `/authentication/tenants/${tenantId}/token?grant_type=password`,
        qs.stringify({
          username,
          password,
        }),
        {
          headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
            Authorization: `Basic ${basicAuthToken}`,
          },
        }
      );
      const { data } = r;
      const { openid, access_token } = data;
      const { sub: userId } = jwt(openid);
      return {
        data,
        service: {
          list: async () => {
            const r = await instance.get(`/catalogue/catalogue/tenants/${tenantId}/services `, {
              headers: {
                Authorization: `Bearer ${access_token}`,
                'X-API-Version': 3,
              },
            });
            const {
              _embedded: { services },
            } = r.data;
            return services;
          },
          get: async (appKey: string) => {
            const r = await instance.get(`/catalogue/catalogue/tenants/${tenantId}/services/${appKey} `, {
              headers: {
                Authorization: `Bearer ${access_token}`,
                'X-API-Version': 3,
              },
            });
            const {
              _embedded: { service },
            } = r.data;
            return service;
          },
        },
        spoke: {
          get: async (type: string) => {
            const r = await instance.get(`/catalogue/catalogue/tenants/${tenantId}/spokes/types/${type}`, {
              headers: {
                Authorization: `Bearer ${access_token}`,
                'X-API-Version': 3,
              },
            });
            const {
              _embedded: { spokes },
            } = r.data;
            return spokes;
          },
        },
        application: {
          list: async () => {
            const r = await instance.get(`/connections/connections/tenants/${tenantId}/applications`, {
              headers: {
                Authorization: `Bearer ${access_token}`,
              },
            });
            return r;
          },
          get: async (appKey: string) => {
            const r = await instance.get(
              `/connections/connections/tenants/${tenantId}/applications/${appKey}/configuration`,
              {
                headers: {
                  Authorization: `Bearer ${access_token}`,
                },
              }
            );

            const {
              _embedded: { configuration },
            } = r.data;
            return configuration;
          },
        },
        company: {
          list: async () => {
            const r = await instance.get(`/customer/customer/tenants/${tenantId}/users/${userId}/companies`, {
              data: null, // needed for this endpoint
              headers: {
                Authorization: `Bearer ${access_token}`,
                'Content-Type': 'application/json',
              },
            });
            const {
              _embedded: { companies },
            } = r.data;

            return companies as [Company];
          },
          get: async (companyUuid: string) => {
            const r = await instance.post(
              `/authentication/tenants/${tenantId}/token?grant_type=token-exchange`,
              qs.stringify({
                context: companyUuid,
                subject_token: openid,
                subject_token_type: 'openid',
              }),
              {
                headers: {
                  Authorization: `Basic ${basicAuthToken}`,
                },
              }
            );
            const { data } = r;
            const { access_token } = data;
            return {
              data,
              send: async (url: string, method: string, data: any) => {
                return instance({
                  baseURL: baseURL + '/connections/connections',
                  url,
                  method,
                  data: qs.stringify(data, { encode: false }),
                  headers: {
                    Authorization: `Bearer ${access_token}`,
                    'Content-Type': 'application/x-www-form-urlencoded',
                  },
                });
              },
              widget: {
                deleteByAppKey: async (appKey: string) => {
                  const r = await instance.delete(
                    `/widget/tenants/${tenantId}/users/${userId}/companies/${companyUuid}/widgets?appKey=${appKey}`,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );

                  return r;
                },
              },
              connection: {
                forApp: async (appKey: string) => {
                  const r = await instance.get(
                    `/connections/connections/tenants/${tenantId}/applications/${appKey}/connections/active`,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );

                  const {
                    _embedded: { connections },
                  } = r.data;
                  return connections;
                },
                list: async () => {
                  const r = await instance.get(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections`,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );

                  const {
                    _embedded: { connections },
                  } = r.data;

                  return connections as [Connection];
                },
                get: async (id: string) => {
                  const r = await instance.get(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections/${id}`,

                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data;
                },
                delete: async (id: string) => {
                  const r = await instance.delete(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections/${id}`,

                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data;
                },
                create: async (appKey: string) => {
                  const r = await instance.post(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections`,
                    qs.stringify({
                      appKey,
                    }),
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  console.log('r ', r);

                  const location = r.request.responseHeaders.Location;
                  const arr = location.split('/');
                  const connection = { id: arr[arr.length - 1] };
                  console.log('create connection ', connection);

                  return connection;
                },
                sendAuth: async (id: string, auth: any) => {
                  const r = await instance.post(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections/${id}/authorization`,
                    qs.stringify(auth),
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                        'Content-Type': 'application/x-www-form-urlencoded',
                      },
                    }
                  );
                  return r.data;
                },
                entities: async (id: string) => {
                  const r = await instance.get(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections/${id}/entities`,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  const {
                    _embedded: { entities },
                  } = r.data;
                  return entities;
                },
                put_entities: async (id: string, entities: any) => {
                  const r = await instance.post(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/connections/${id}/entities`,
                    entities,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data;
                },
              },

              workflow: {
                update: async (workflowId: string, activityId: string, stepId: string) => {
                  const r = await instance.put(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/workflow/${workflowId}?activityId=${activityId}&stepId=${stepId}`,
                    null,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data.workflow;
                },
                resume: async (appKey: string) => {
                  const r = await instance.get(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/workflow/resume?appKey=${appKey}&outcome=connect`,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data.workflow;
                },
                reconnect: async (appKey: string) => {
                  const r = await instance.post(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/workflow?appKey=${appKey}&outcome=reconnect`,
                    null,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data.workflow;
                },
                create: async (appKey: string) => {
                  const r = await instance.post(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/workflow?appKey=${appKey}&outcome=connect`,
                    null,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data.workflow;
                },
                get: async (appKey: string) => {
                  const r = await instance.get(
                    `/connections/connections/tenants/${tenantId}/company/${companyUuid}/workflow`,
                    {
                      headers: {
                        Authorization: `Bearer ${access_token}`,
                      },
                    }
                  );
                  return r.data;
                },
              },
            };
          },
        },
      };
    },
  };
};

export default client;
