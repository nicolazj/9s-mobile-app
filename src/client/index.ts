import axios, { AxiosInstance } from 'axios';
import qs from 'qs';
import jwt from 'jwt-decode';
import user from './user';
import company from './company';
import config from './config';
import { Buffer } from 'buffer';
import { ClientConfig, UserAuth, CompanyAuth, LoginPayload, User } from '../types';

class APIClient {
  private config: ClientConfig;
  private userAuth?: UserAuth;
  private companyAuth?: CompanyAuth;

  private instance: AxiosInstance;
  constructor(config: ClientConfig) {
    const { baseURL, appKey, appSecret } = config;
    this.instance = axios.create({
      baseURL,
      headers: {
        Authorization: `Basic ${Buffer.from(`${appKey}:${appSecret}`).toString('base64')}`,
      },
    });
    this.config = config;
  }
  login = async (payload: LoginPayload) => {
    const { tenantId } = this.config;

    const config = {
      method: 'POST',
      url: `/authentication/tenants/${tenantId}/token?grant_type=password`,
      data: qs.stringify(payload),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
    };
    const r = await this.instance(config);
    const { data } = r;

    const { openid } = data;
    const { sub: userId } = jwt(openid);
    this.userAuth = { ...data, userId } as UserAuth;
    return this.userAuth;
  };

  auth = async (companyUuid: string) => {
    if (!this.userAuth) {
      throw new Error('no user auth');
    }
    const { tenantId } = this.config;
    const { openid } = this.userAuth;

    const r = await this.instance.post(
      `/authentication/tenants/${tenantId}/token?grant_type=token-exchange`,
      qs.stringify({
        context: companyUuid,
        subject_token: openid,
        subject_token_type: 'openid',
      })
    );
    const { data } = r;

    this.companyAuth = { ...data, companyUuid } as CompanyAuth;

    return data;
  };

  get user() {
    return this.userAuth ? user(this.config, this.userAuth) : null;
  }
  get company() {
    return this.companyAuth ? company(this.config, this.companyAuth) : null;
  }
}

export default new APIClient(config);
