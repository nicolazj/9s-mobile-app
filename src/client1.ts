import axios from 'axios';
import jwt from 'jwt-decode';
import qs from 'qs';
import { Buffer } from 'buffer';

interface Company {
  companyName: string;
  companyUuid: string;
  industryUuid: String;
}
interface Connection {
  appKey: string;
  createdAt: string;
  createdBy: string;
  id: string;
  status: string;
}

interface ClientConfig {
  baseURL: string;
  tenantId: string;
  appKey: string;
  appSecret: string;
}

const Client = ({ baseURL, tenantId, appKey, appSecret }: ClientConfig) => {
  const instance = axios.create({
    baseURL,
  });

  const basicAuthToken = Buffer.from(`${appKey}:${appSecret}`).toString('base64');
  return {
    login: async ({ username, password }: { username: string; password: string }) => {
      const r = await instance.post(
        `/authentication/tenants/${tenantId}/token?grant_type=password`,
        qs.stringify({
          username,
          password,
        }),
        {
          headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
            Authorization: `Basic ${basicAuthToken}`,
          },
        }
      );
      const { data } = r;
      const { openid, access_token } = data;
      const { sub: userId } = jwt(openid);

      const user = {
        me: async () => {
          const r = await instance.get(`/customer/customer/tenants/${tenantId}/users/${userId}`, {
            data: null,
            headers: {
              Authorization: `Bearer ${access_token}`,
              'Content-Type': ' application/json',
            },
          });
          return r.data;
        },
        get: async (userId: string) => {
          const r = await instance.get(`/customer/customer/tenants/${tenantId}/users/${userId}`, {
            data: null,
            headers: {
              Authorization: `Bearer ${access_token}`,
              'Content-Type': ' application/json',
            },
          });
          return r.data;
        },
      };
      return {
        data,
        user,
        service: {
          list: async () => {
            const r = await instance.get(`/catalogue/catalogue/tenants/${tenantId}/services `, {
              headers: {
                Authorization: `Bearer ${access_token}`,
                'X-API-Version': 3,
              },
            });
            const {
              _embedded: { services },
            } = r.data;
            return services;
          },
          get: async (appKey: string) => {
            const r = await instance.get(`/catalogue/catalogue/tenants/${tenantId}/services/${appKey} `, {
              headers: {
                Authorization: `Bearer ${access_token}`,
                'X-API-Version': 3,
              },
            });
            const {
              _embedded: { service },
            } = r.data;
            return service;
          },
        },
        spoke: {
          get: async (type: string) => {
            const r = await instance.get(`/catalogue/catalogue/tenants/${tenantId}/spokes/types/${type}`, {
              headers: {
                Authorization: `Bearer ${access_token}`,
                'X-API-Version': 3,
              },
            });
            const {
              _embedded: { spokes },
            } = r.data;
            return spokes;
          },
        },
      };
    },
  };
};

export default Client({
  baseURL: 'https://api.9spokes.io/dev02',
  tenantId: '00000000-0000-0005-5555-555555555555',
  appKey: 'mobileSmudgeDEV02',
  appSecret: 'zYfCuhQnBz5rvv7nR9s9JZ2AXY49uVVhYYN2BExF',
});
