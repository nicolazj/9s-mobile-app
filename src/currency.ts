export const currencyMaps = [
  { currency: 'GBP', symbol: 'Â£' },
  { currency: 'NZD', symbol: '$' },
];
export const getSymbol = (currency: string) => {
  const r = currencyMaps.find(m => m.currency === currency);
  return r ? r.symbol : '$';
};
