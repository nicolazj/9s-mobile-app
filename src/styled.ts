import * as styledComponents from 'styled-components';
import get from 'lodash.get';
const {
  default: styled,
  css,
  keyframes,
  ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<IThemeInterface>;

export interface IThemeInterface {
  color: {
    main: string;
  };
}

export const th = (...props: string[]) => ({ theme }: { theme: IThemeInterface }) => {
  for (let i = 0; i < props.length; i++) {
    let res = get(theme, props[i]);
    if (res) return res;
  }
};

export default styled;
export { css, keyframes, ThemeProvider };
