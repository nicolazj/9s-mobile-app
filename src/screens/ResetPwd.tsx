import { Field, Formik } from 'formik';
import React from 'react';
import { Image, View } from 'react-native';
import { NavigationScreenProp } from 'react-navigation';

import agent from '../agent';
import Button from '../components/Button';
import Link from '../components/Link';
import { FormDesc, FormikTextInput, FormTitle } from '../formik';
import log from '../logging';
import * as P from '../primitives';
import { SCREENS } from '../routes/constants';
import { alert } from '../services/Alert';
import { object, username } from '../validations';

interface Props {
  navigation: NavigationScreenProp<any, any>;
}

const ResetPwd: React.FC<Props> = ({ navigation }) => {
  const [done, setDone] = React.useState(false);
  const onPress = async (values: { email: string }) => {
    try {
      await agent.public.password.reset(values.email);
      setDone(true);
    } catch (err) {
      log('reset error', err);
      alert('Reset password failed', 'try again later');
    }
  };

  return (
    <P.Container>
      <P.SafeArea>
        {done ? (
          <P.Container hasPadding hcenter vcenter>
            <Image
              style={{ width: 150, height: 150 }}
              source={{ uri: 'https://via.placeholder.com/150' }}
            />
            <FormTitle>Email sent!</FormTitle>
            <P.Text>Check your inbox for a link to </P.Text>
            <P.Text> reset your password</P.Text>
            <Link
              title="Done"
              onPress={() => {
                navigation.navigate(SCREENS[SCREENS.SIGN_IN]);
              }}
              style={{ margin: 20 }}
            />
          </P.Container>
        ) : (
          <P.Container hasPadding>
            <Formik
              initialValues={{
                email: '',
              }}
              validationSchema={object().shape({
                email: username,
              })}
              onSubmit={onPress}
            >
              {({ handleSubmit }) => (
                <View>
                  <FormTitle style={{ marginBottom: 30 }}>
                    Reset your password
                  </FormTitle>
                  <FormDesc>
                    No problem! Enter your email address and we will send you a
                    link to reset your password
                  </FormDesc>
                  <Field
                    name="email"
                    component={FormikTextInput}
                    placeholder="Email"
                  />
                  <Button title="Reset password" onPress={handleSubmit} />
                </View>
              )}
            </Formik>
          </P.Container>
        )}
      </P.SafeArea>
    </P.Container>
  );
};
export default ResetPwd;
