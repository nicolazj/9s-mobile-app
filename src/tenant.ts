import { IThemeInterface } from './styled';
export const theme: IThemeInterface = {
  color: {
    main: '#39D5A7',
    header: '#272149',
    grey: '#aaa',
    view: { bg: '#f6f6f6' },
  },
};
