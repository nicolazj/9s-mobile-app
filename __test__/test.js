import React from 'react';
import ResetPwd from '../src/screens/ResetPwd';
import renderer from 'react-test-renderer';
import tenant from '../src/tenant';
import { ThemeProvider } from '../src/styled';
const createTestProps = props => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});
test('renders correctly', () => {
  const props = createTestProps({});
  const tree = renderer
    .create(
      <ThemeProvider theme={tenant.theme}>
        <ResetPwd {...props} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
