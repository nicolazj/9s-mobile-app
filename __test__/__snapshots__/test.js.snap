// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f6f6f6",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      hasPadding={true}
      style={
        Array [
          Object {
            "backgroundColor": "#f6f6f6",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 0,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "System",
                "fontSize": 56,
                "fontWeight": "400",
                "lineHeight": 95,
                "marginBottom": 40,
                "marginLeft": 40,
                "marginRight": 40,
                "marginTop": 40,
                "textAlign": "center",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Reset your password
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#aaa",
                "fontFamily": "System",
                "fontSize": 28,
                "fontWeight": "400",
              },
            ]
          }
        >
          No problem! Enter your email address and we will send you a link to reset your password
        </Text>
        <View
          style={
            Array [
              Object {
                "height": 120,
                "marginBottom": 10,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 10,
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Email"
            placeholderTextColor="#aaa"
            rejectResponderTermination={true}
            selectionColor="#39D5A7"
            style={
              Array [
                Object {
                  "borderColor": "#aaa",
                  "borderRadius": 5,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "fontSize": 28,
                  "paddingBottom": 32,
                  "paddingLeft": 32,
                  "paddingRight": 32,
                  "paddingTop": 32,
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#39D5A7",
              "borderColor": "#39D5A7",
              "borderRadius": 5,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 96,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
              "opacity": 1,
              "paddingBottom": 30,
              "paddingLeft": 30,
              "paddingRight": 30,
              "paddingTop": 30,
              "width": "100%",
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontSize": 28,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#fff",
                    "textAlign": "center",
                  },
                ]
              }
            >
              Reset password
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
